plugin/SqueezeESP32/HTML/EN/plugins/SqueezeESP32/settings/player.html

```diff
@@ -35,6 +35,13 @@
 			<input type="text" class="stdedit" name="pref_artwork_y" id="artwork_y" value="[% pref_artwork.y %]" size="2">
 		[% END %]
 
+		[% WRAPPER setting title="PLUGIN_SQUEEZEESP32_LED_VU" desc="PLUGIN_SQUEEZEESP32_LED_VU_DESC" %]
```

Comment:
We should make that optional with a [% of pref.xxxx %] adn find a way for the plugin to know if there is a led_vu

Reply:
I do intend to change this from a submit to java-script function.  When I do, I'll have a look at this too.  Mind, like the artwork function, all it does is send a slimproto message that will be ignored by any player except for a Squeezelite-esp32 with an initialized led_vu.

Reply:
But we can easily add data to the current SETD message, it will not break current player's compatibility and the plugin will know that there is a led strip. See
```
# Allow the player to define it's display width (and probably more)
sub playerSettingsFrame {
	my $client   = shift;
	my $data_ref = shift;

	my $value;
	my $id = unpack('C', $$data_ref);

	# New SETD command 0xfe for display width & height
	if ($id == 0xfe) {
		$value = (unpack('Cn', $$data_ref))[1];
		if ($value > 100 && $value < 400) {
			$prefs->client($client)->set('width', $value);

			my $height = (unpack('Cnn', $$data_ref))[2];
			$prefs->client($client)->set('height', $height || 0);

			$client->display->modes($client->display->build_modes);
			$client->display->widthOverride(1, $value);
			$client->update;

			main::INFOLOG && $log->is_info && $log->info("Setting player $value" . "x" . "$height for ", $client->name);
		}
	}

	$client->SUPER::playerSettingsFrame($data_ref);
}
```

Reply:
strip length is probably all we need, but I'm still thinking along the screen2 route so maybe a second width and height.   

You suggest adding additional fields to -2(0xfe), or use a different frame id like -3(0xfd).  (have not located a comprehensive list, only ones i see used are 0-6 on SB2 and -2(0xfe) by squeezeSlave/now squeezeESP32).  Later is potentially more risky, but if we want to separate led_vu and display modules, a different id may be needed.